lib/utils/index.js

```diff
@@ -645,21 +645,11 @@ class Utils {
   }
   
   static stringifyObject(objects) {
-    if (Array.isArray(objects)) {
-      return objects.map(obj => {
-        if (!Array.isArray(obj) && Utils.isObject(obj)) {
-          return inspect(obj);
-        }
-
-        return obj;
-      });
-    }
-
     if (Utils.isObject(objects)) {
-      return inspect(objects);
+      return JSON.stringify(inspect(objects));
```

Comment:
this change isn't necessary, the `inspect()` always returns a string. The actual issue was fixed by the previous change in `assertion.js`

Reply:
The previous change did not fix the issue completely, for nested objects template literal returns `"[object Object]"`.

Reply:
Removed `JSON.stringify`

Reply:
Ok, inspect has a default nesting level of 3 I think, which can be configured. But in any case, JSON.stringify doesn't solve that problem, since the result from inspect is already a string. Or am I misunderstanding the issue?

It would solve it if you'd remove the inspect completely, but then we'd have the ugly JSON serialization with all the double-quotes, which I was tried to avoid with using inspect.

Reply:
You got the issue correctly. I did not realise inspect returns a string. I was trying to use JSON serialisation but that did work because we have objects with circular dependencies. Inspect can work with circular dependencies and the output is cleaner. We do not need to print the object to infinite depth, 3 is fine in my opinion.